content/fundamentals/provider-scopes.md

```diff
@@ -103,6 +103,42 @@ export class CatsService {
 
 You then configure your `context` value (in the `GraphQLModule`) to contain `request` as its property.
 
+#### Inquirer provider
+
+If you want get the class where a provider was constructed, for instance in logging or metrics providers, you can inject the `INQUIRER` token.
+
+```typescript
+import { Inject, Injectable, Scope } from '@nestjs/common';
+import { INQUIRER } from '@nestjs/core';
+
+@Injectable({ scope: Scope.TRANSIENT })
+export class HelloService {
+  constructor(@Inject(INQUIRER) private parentClass: object) {}
+
+  sayHello(message: string) {
+    console.log(`${this.parentClass?.constructor?.name}: ${message}`);
+  }
+}
+```
+
+```typescript
+import { Injectable } from '@nestjs/common';
+import { HelloService } from './hello.service';
+
+@Injectable()
+export class AppService {
+  constructor(private helloService: HelloService) {}
+
+  getRoot(): string {
+    this.helloService.sayHello('My name is getRoot');
+
+    return 'Hello world!';
+  }
+}
+```
+
+In the example above when `AppService.getRoot` is called, `'AppService: My name is getRoot'` will be logged to the console.
```

Comment:
```suggestion
In the example above when `AppService#getRoot` is called, `"AppService: My name is getRoot"` will be logged to the console.
```

Reply:
Thanks for the review!

I'm not familiar with this hash formatting and couldn't find it elsewhere in the docs, but happy to pull it down either way.